driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -103,14 +107,52 @@ public ChangeStreamDocument(@BsonProperty("operationType") final String operatio
         this.fullDocument = fullDocument;
         this.clusterTime = clusterTime;
         this.operationTypeString = operationTypeString;
-        this.operationType = OperationType.fromString(operationTypeString);
+        this.operationType = operationTypeString == null ? null : OperationType.fromString(operationTypeString);
```

Comment:
fromString does not take null; note that it is used in the OperationTypeCodec

Reply:
I am glad we are not passing `null` to `OperationType.fromString`, even though it (arguably, incorrectly) has the code to handle `null`s.